### ISSUE TITLE
Server not listening on ipv4 3688 websocket port on FreeBSD

### ISSUE THREAD

**Issue by ejurgensen** [owntone/owntone-server#1255]
@chme and @luusl, thanks for fixing the websocket issue so quickly. Unfortunately I ran into another websocket issue while testing the new release: On FreeBSD, the server isn't listening on both ipv4 and ipv6 port 3688, as would be expected when ipv6 is enabled. It only listens on ipv6. I think this is due to IPV6_V6ONLY being enabled by default on FreeBSD.

I've googled for solutions for this, but didn't find any. Perhaps you can? Or can find another project that uses libwebsockets and has solved this?

If not, then I'm tempted to just drop websocket ipv6 support for FreeBSD, so that at least ipv4 works.

Btw this is probably an old issue, maybe it has always been like this. I'm not sure I have ever tested this before.

**Comment by luusl** [comment]
For me it works on Linux and macOS. You can try to explicitly disable the ipv6only option by adding `LWS_SERVER_OPTION_IPV6_V6ONLY_MODIFY` to the info struct in `websocket_init()`, e.g.
```C
  if (!cfg_getbool(cfg_getsec(cfg, "general"), "ipv6"))
    info.options |= LWS_SERVER_OPTION_DISABLE_IPV6;
  else
    info.options |= LWS_SERVER_OPTION_IPV6_V6ONLY_MODIFY;
```
then libwebsockets will explicitly call `
setsockopt(sockfd, IPPROTO_IPV6, IPV6_V6ONLY, (const void*)&value, sizeof(value))` where `int value = 0;`.

**Comment by ejurgensen** [comment]
That looks perfect. I will give at a try later, but I would be surprised if it doesn't solve the issue.

**Comment by ejurgensen** [comment]
Closed with commit 1a56255
